internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -89,19 +90,100 @@ Overview of the connection process
 	MsSQL->Driver: Login Response
 	Driver->Secretless: Login success or failure
 	Secretless->Client: Login Response (Premade)
- */
+*/
 
 // SingleUseConnector is used to create an authenticated connection to an MSSQL target
 type SingleUseConnector struct {
-	backendConn net.Conn
-	clientConn  net.Conn
-	logger      log.Logger
+	backendConn       net.Conn
+	clientConn        net.Conn
+	logger            log.Logger
+	newMSSQLConnector types.NewMSSQLConnectorFunc
+	readPrelogin      types.ReadPreloginFunc
+	writePrelogin     types.WritePreloginFunc
+	newTdsBuffer      types.NewTdsBufferFunc
+}
+
+// NewSingleUseConnector creates a new SingleUseConnector
```

Comment:
Maybe useful to add comment that `NewSingleUseConnector` creates a production version of a SingleUseConnector?